index.py

```diff
@@ -129,9 +131,30 @@ def extract_git_metadata(path):
     return write_date, create_date, authors
 
 
-def discover_articles(root):
+def discover_articles(root, directories):
     """ Discover all articles under ``root``. """
 
+    articles = {}
+    rejected_files = []
+    for path in root.glob('**/*.md'):
+        for i in directories.glob('**/*.md'):
+            rejected_files.append(i)
```

Comment:
Pas besoin de recalculer la liste pour chaque fichier
```suggestion
    for i in directories.glob('**/*.md'):
        rejected_files.append(i)
    for path in root.glob('**/*.md'):
```